### PR TITLE
Added MDC and log statements

### PR DESCRIPTION
**Pull request by sivakaruna**
# Description

This PR includes the following proposed change(s):

- DF-328

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

Local tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: digitalforms-api/src/main/java/ca/bc/gov/open/pssg/rsbc/digitalforms/controller/ApplicationFormController.java [hunk 1]
```diff
@@ -44,64 +47,100 @@
 	@Autowired
 	ApplicationFormService service;
 
-	@GetMapping(value = "/{formType}/{GUID}/application", produces = DigitalFormsConstants.JSON_CONTENT)
+	Logger logger = LoggerFactory.getLogger(ApplicationFormController.class);
+
+	@GetMapping(value = "/{formType}/{GUID}/application/{requestGuid}", produces = DigitalFormsConstants.JSON_CONTENT)
```
Comment (shaunmillargov) [inline digitalforms-api/src/main/java/ca/bc/gov/open/pssg/rsbc/digitalforms/controller/ApplicationFormController.java:52]: If I'm understanding this correctly, the requestGuid param is the key we use to track the logging for this request. If so, could you please change the name of this variable to correlationId? This is the term most widely used and understood for this value. Thanks Siva.

File: digitalforms-api/src/main/java/ca/bc/gov/open/pssg/rsbc/digitalforms/util/DigitalFormsConstants.java [hunk 2]
```diff
@@ -45,4 +45,8 @@ private DigitalFormsConstants() {
 	// rest response media type
 	public static final String JSON_CONTENT = "application/json";
    
+	// MDC constants
+	public static final String REQUEST_GUID = "request.guid";
```
Comment (shaunmillargov) [inline digitalforms-api/src/main/java/ca/bc/gov/open/pssg/rsbc/digitalforms/util/DigitalFormsConstants.java:49]: Change the naming here too please.

File: digitalforms-api/src/main/resources/logback-splunk.xml [hunk 3]
```diff
@@ -5,7 +5,7 @@
     ​
     <!-- You can override this to have a custom pattern -->
     <property name="CONSOLE_LOG_PATTERN"
-              value="%clr(%d{yyyy-MM-dd HH:mm:ss.SSS}){faint} %clr(${LOG_LEVEL_PATTERN:-%5p}) %clr(${PID:- }){magenta} %clr(---){faint} %clr([%15.15t]){faint} %clr(%-40.40logger{39}){cyan} %clr(:){faint} %m%n${LOG_EXCEPTION_CONVERSION_WORD:-%wEx}"/>
+              value="%clr(%d{yyyy-MM-dd HH:mm:ss.SSS}){faint} %clr(${LOG_LEVEL_PATTERN:-%5p}) {%X{request.guid}  %X{request.formtype} %X{request.endpoint}} %clr(${PID:- }){magenta} %clr(---){faint} %clr([%15.15t]){faint} %clr(%-40.40logger{39}){cyan} %clr(:){faint} %m%n${LOG_EXCEPTION_CONVERSION_WORD:-%wEx}"/>
```
Comment (shaunmillargov) [inline digitalforms-api/src/main/resources/logback-splunk.xml:8]: and here

File: digitalforms-api/src/main/java/ca/bc/gov/open/pssg/rsbc/digitalforms/controller/ApplicationFormController.java (hunk 1 above)
Comment (sivakaruna) [reply digitalforms-api/src/main/java/ca/bc/gov/open/pssg/rsbc/digitalforms/controller/ApplicationFormController.java]: updated

File: digitalforms-api/src/main/resources/logback-splunk.xml (hunk 3 above)
Comment (sivakaruna) [reply digitalforms-api/src/main/resources/logback-splunk.xml]: updated

File: digitalforms-api/src/main/java/ca/bc/gov/open/pssg/rsbc/digitalforms/util/DigitalFormsConstants.java (hunk 2 above)
Comment (sivakaruna) [reply digitalforms-api/src/main/java/ca/bc/gov/open/pssg/rsbc/digitalforms/util/DigitalFormsConstants.java]: updated

**Pull request merged**